### DISCUSSION TITLE
Directus 9: How do I update user with custom api

### DISCUSSION THREAD

**Discussion by harshalone** [directus/directus#9815] [Archive]
I am verifying user with a verification code.

After verification I need to update user status to active how do I do it?

In the documentation [here](https://docs.directus.io/extensions/api-endpoints/) it says 

- services — All API internal services.
- exceptions — API exception objects that can be used to throw "proper" errors.
- database — Knex instance that is connected to the current database.
- getSchema — Async function that reads the full available schema for use in services
- env — Parsed environment variables.
- logger — Pino (opens new window)instance.

so lets say I have to uddate a user status. How do I exactly do that?

`users.update('user_id', 'KEY', "VALUE"); ? `

Please help

**Comment by vnicolasdev** [comment]
You can use database knex instance as following:
`await database('users').update({id: user_id, status: 'active'})`

**Comment by rijkvanzanten** [comment]
Or you can use the services you mentioned, for example:

```js
const { ItemsService } = services;

const usersService = new ItemsService('users', { schema });

await usersService.updateOne(user_id, { status: 'active' });
```
